### ISSUE TITLE
CIP chirality not matching all cases from CIPValidation suite

### ISSUE THREAD

**Issue by shubbey** [rdkit/rdkit#5357]
I was looking at the molecules here: https://cipvalidationsuite.github.io/ValidationSuite/

In RDKit, several of these are not generating the correct chirality.    For example, VS298:
```
from rdkit import Chem
rdmol = Chem.MolFromSmiles("Cl[C@H]1C[C@]2(C1)C[C@H](C2)Cl")
Chem.AssignStereochemistry(rdmol)
for rda in rdmol.GetAtoms():
    if rda.HasProp("_CIPCode"):
        print(rda.GetProp("_CIPCode"))
```

It is my understanding that this implementation is ported from Centres.  Inputting the same smiles into their tool here: https://www.simolecule.com/cdkdepict/depict.html

Gives the expected result.  So my question is, am I calling this incorrectly or is there a glitch here? Many of the examples in the Validation Suite fail similarly.  Note that my actual implementation is through mol files in C++ but I get the same results as above.  I've tried different sanitation pre-processing (although I wouldn't expect this to be necessary) and also different stereochemistry assignment (eg since I am importing from a 3D file, I use assignStereochemistryFrom3D prior to assignCIPLabels).  What am I doing wrong?

Thanks!

**Comment by greglandrum** [comment]
@shubbey `AssignStereochemistry()` still calls the RDKit's old, and imperfect, CIP labelling code. The new code is the function `rdkit.Chem.rdCIPLabeler.AssignCIPLabels()`. You also need to disable sanitization when parsing the molecule so that the old code doesn't remove the input stereo:
```
In [6]: rdmol = Chem.MolFromSmiles("Cl[C@H]1C[C@]2(C1)C[C@H](C2)Cl", sanitize=False)

In [7]: Chem.SanitizeMol(rdmol)
Out[7]: rdkit.Chem.rdmolops.SanitizeFlags.SANITIZE_NONE

In [8]: rdCIPLabeler.AssignCIPLabels(rdmol)

In [9]: for rda in rdmol.GetAtoms():
   ...:     if rda.HasProp("_CIPCode"):
   ...:         print(rda.GetProp("_CIPCode"))
R
S
R
```

I'm working on making the new stereo code the default for the next release.

**Comment by shubbey** [comment]
Thank you, Greg!  

I posted some erroneous code above--  I was calling AssignCIPLabels() in both python and C++ but as you helpfully noted, the sanitation was causing problems.

My actual use case is loading a molecule for which I have a 3D structure.  I looked at the mol loader, MolDataStreamToMol, and cobbled together this minimal code that seems to do the trick:

```
        // after loading atoms and bonds into rdmol...
	rdmol.updatePropertyCache(false);
	MolOps::assignChiralTypesFrom3D(rdmol);
	CIPLabeler::assignCIPLabels(rdmol);
```
